src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -129,24 +130,54 @@ private static int KMPSearch(byte[] pattern, byte[] bytes)
                     }
                 }
             }
+
             return -1;
         }
 
-        private static void SearchAndReplace(byte[] array, byte[] searchPattern, byte[] patternToReplace, string appHostSourcePath)
+        private static unsafe void SearchAndReplace(
+            MemoryMappedViewAccessor accessor,
+            byte[] searchPattern,
+            byte[] patternToReplace,
+            string appHostSourcePath)
         {
-            int offset = KMPSearch(searchPattern, array);
-            if (offset < 0)
+            byte* bytes = null;
```

Comment:
nit: bytes can be declared at first use. It is not access outside the try {}.

Reply:
fixed